ucfcs/GrowthPlus
# Julio/cat count

This PR implements:
- Increasing the category count based on lesson for the child schema so it can be used for progress bars.
- Clearing out the deplyomentDropDown file from git cache so it stops tracking and changing it every time.
- Refactoring flashcards, quizzes and games to keep track of number of correct questions.
- Refactoring flashcards to implement a realm listener so that the UI doesn't update faster than realm causing it to use old data.
- Updating the state of each roadmap based on game completion. 

To test:
- Uncomment line 68 in MainActivity.java, this will import a sample child with mostly completed level 1 and level 1 game locked.
- Clear out data from emulator and do a fresh build.
- You may or may not notice the sample child "lili" at first, create a parent and go back to landing page, lili should be there and create a another child. 
- Child lili has all but last lesson completed and game lock in level 1, the created child has everything unlock except last game in level 4. 
- Select lili and play last lesson, pass it and notice the game is unlocked, play game, pass it, you now should now see level 2 unlock and starting at tile one. The state of lili will reset every time because we are using realm importer when we go back to landing page, this is expected.  
- Do the same for the child you created, play the last lesson and game and notice the state of the roadmap chances and everything is completed. 
- Open the realm file and inspect that the category count review is increased by one in child schema, because we are artificially adding data when creating and importing children, no other category will update but the functionality is there. 
- Flashcard, quiz and game have a numCorrect counter to keep track of the number of correct answers, this resets every time. 
- Play around with the tiles, make sure nothing breaks.
